### DISCUSSION TITLE
Why isn't there an option to log to a file?

### DISCUSSION THREAD

**Discussion by kierans** [requarks/wiki#6766] [Help / Questions] [answered]
Apologies if this has been asked before, but I've searched around and not found the answer.

I'm wondering why there isn't an option in the config to log to a file instead of/with the console/stdout. My use case is that I'm hosting in a multiple Docker environment and I want to stream container logs to my log utility (eg: CloudWatch)

My current solution is to create my own Docker image of WikiJS and setup command redirection to log to a file which is then mapped to the host system and read by the logs agent. However given Winston is the logging lib used, it would be easy enough to add file logging to the server, with an opt in config item to enable it.

This is something I could take a look at doing if there's appetite for it, however there may be a reason or a deliberate choice not to log to a file.

**Comment by NGPixel** [comment, marked as answer]
You're doing it wrong? The container logs can be streamed to any output of your choice... Docker has many logging drivers, including to file, json, syslog, etc.

e.g. a CloudWatch log driver:
https://docs.docker.com/config/containers/logging/awslogs/
